src/Compiler/CompilationResult.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+/**
+ * SCSSPHP
+ *
+ * @copyright 2012-2020 Leaf Corcoran
+ *
+ * @license http://opensource.org/licenses/MIT MIT
+ *
+ * @link http://scssphp.github.io/scssphp
+ */
+
+namespace ScssPhp\ScssPhp\Compiler;
+
+use ScssPhp\ScssPhp\Util;
+use ScssPhp\ScssPhp\Exception\CompilerException;
+
+/**
+ * Compiler environment
+ *
+ */
+class CompilationResult
+{
+    protected static $resolvedImport = [];
```

Comment:
why static ?

Reply:
Static is used here for performance issue accros multiples file compilation sharing mixins or functions libs.

My testing use case is 4 .scss compilation: the full BS4 as a first and then 3 others using parts of BS4 (like variables, mixins..). Without the static a lot of validation are occuring multiple times and this is costly.

Using this static allows a total cost for this full check that is quite acceptable.

I guess to be bulletproof, the `checkValid()` method should have an optional `$forceRefresh` argument to empty the static and force a full re-check?

Reply:
if we need to share some state between compilations, I think this should live in the Compiler class rather than being in a static property in CompilationResult. Each compilation has its own CompilationResult.

btw, using a static property here would even leak the shared state between multiple Compiler instances, which might have different configuration.

Reply:
This is purely an internal storage stuff and the case of multiple compiler instances with different configurations is taken in account through the hash, to reuse previous check only if 
https://github.com/scssphp/scssphp/pull/270/commits/f57610d92bcfbbf50e387d64a950e7d95ff037c1#diff-a95b369f4118d2d8d85bb94f822b8d5ca7e92c93a9c0d533e43742b8f3e6e742R277

In my test case indeed each scss compilation occurs with a new Compiler Instance - but I guess I could now use a static instance to reuse it accross several compilations.

Alternatively I can remove the static optimization from the CompilerResult class, delivering something ready to use and 100% clean - even if not the fastest implementation.
And besides that add an option to the `getImportedFiles()` method to return the full array and not just one column.

Then it allows users to implement their own checking method, possibly faster with some compromise on compilation isolation.

For me this is making a difference:
- the fast checking method, with only the timestamps is around **~10ms** for the whole scss compilations
- with a deep checking with non-static array optimisation : **~85ms**
- with a deep checking with static array optimisation :  **~45ms**

But I also have to admit the difference is not as big as I had with my first implementation where i measured a total of  ~250ms with a full check (but I can't figure out what I changed since).

Let's go for this compromise?

Reply:
so, no more static, just a local variable